mod/item.php

```diff
@@ -175,6 +175,19 @@ function item_post(&$a) {
 	$app         = ((x($_REQUEST,'source'))      ? strip_tags($_REQUEST['source'])  : '');
 	$extid       = ((x($_REQUEST,'extid'))       ? strip_tags($_REQUEST['extid'])  : '');
 
+	// Check for multiple posts with the same message id (when the post was created via API)
+	if (($message_id != '') AND ($profile_uid != 0)) {
```

Comment:
The whole functionality in include/items.php and mod/item.php needs to be reworked completely. I want to replace it with an item class to have centralized the needed SQL queries. (To be able to replace them easily when we reworked the database structure)

So the time of these mammoth size functions is really limited.